pkg/src/src/main/scala/edu/berkeley/cs/amplab/sparkr/RRDD.scala

```diff
@@ -312,6 +314,27 @@ private class StringRRDD[T: ClassTag](
   lazy val asJavaRDD : JavaRDD[String] = JavaRDD.fromRDD(this)
 }
 
+private class BufferedStreamThread(in: InputStream, name: String) extends Thread(name) {
+  val ERR_BUFFER_SIZE = 100
+  val lines = new Array[String](ERR_BUFFER_SIZE)
+  var lineIdx = 0
+  override def run() {
+    for (line <- Source.fromInputStream(in).getLines) {
```

Comment:
The trouble is `output` here is an `Iterator[String]`. If we want to take two passes on the iterator (one for println and one for takeRight) we'll need to convert it to an array and I don't want to buffer the entire `stderr` to an array.